seametrics/softmin_score/softmin_score.py

```diff
@@ -0,0 +1,196 @@
+        resized_mask = np.zeros((actual_height, actual_width), dtype=mask.dtype)
+        scale_y = actual_height / mask.shape[0]
+        scale_x = actual_width / mask.shape[1]
+        for i in range(actual_height):
+            for j in range(actual_width):
+                orig_y = int(i / scale_y)
+                orig_x = int(j / scale_x)
+                resized_mask[i, j] = mask[orig_y, orig_x]
+        mask = resized_mask
+    # Place mask in image mask
+    image_size_mask[abs_top:abs_bottom, abs_left:abs_right] = mask * index
+
+    return image_size_mask
+
+
+def create_image_mask(annotations, image_size):
+    """
+    Create an image mask from a list of mask annotations.
+    
+    Args:
+    - annotations: List of annotations, where each annotation is a dictionary with keys 'bounding_box', 'mask', and 'index'.
+    - image_size: Tuple (height, width) representing the size of the image mask to be created.
+    
+    Returns:
+    - image_mask: NumPy array representing the image mask with object indexes.
+    """
+    height, width = image_size
+    image_mask = np.zeros((height, width), dtype=np.uint8)   
+    for annotation in annotations:
+        bbox = annotation['bounding_box']
+        mask = annotation['mask']
+        index = label_to_id[annotation['label']]
+        object_mask_full = transform_mask_to_image_size(image_size, bbox, mask, index)
+        image_mask[object_mask_full>0] = index
+
+    return image_mask
+
+def normalize_mask(mask, obj_id):
+    # Find pixels belonging to the current object
+    y_indices, x_indices = np.where(mask == obj_id) 
+    # Calculate the bounding box
+    x_min, x_max = x_indices.min(), x_indices.max()
+    y_min, y_max = y_indices.min(), y_indices.max()
+
+    # Normalize the bounding box
+    top, left = y_min / mask.shape[0], x_min / mask.shape[1]
+    width, height = (x_max - x_min) / mask.shape[1], (y_max - y_min) / mask.shape[0]
+    normalized_bb = [left, top, width, height]
+
+    # Create a binary mask for the object
+    obj_mask = (mask[y_min:y_max+1, x_min:x_max+1] == obj_id).astype(np.uint8)
+
+    return normalized_bb, obj_mask
+
+
+def softmin_output_to_fo_format(mask):
+    objects = np.unique(mask)
+    objects = objects[objects != 0]  # Exclude background
+
+    fo_annotations = []
+    for obj_id in objects:
+
+        normalized_bb, obj_mask = normalize_mask(mask, obj_id)
+
+        annotation = fo.Detection(bounding_box=normalized_bb, mask=obj_mask, label='softmin_error')
+        fo_annotations.append(annotation)
+
+    return fo_annotations
+
+
+all_softmin_scores = []
+
+time0 = time.time()
+for sample in tqdm(dataset_view):
+    image_filepath = sample.filepath
+    # image = Image.open(image_filepath).convert('RGB')
+    image = np.array(Image.open(image_filepath))
+    image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+    image = Image.fromarray(image)
+
+    pixel_values = image_transform(image=np.array(image))["image"]
```

Comment:
I think we should either stick with numpy or pillow. We are reading with pillow, then to numpy/cv2, then back to pillow and then back to np array haha.

Reply:
This was originally only PIL. but the problem is, readin the images and converting the mode from 'L' to 'RGB' doesn't work somehow, that is why the conversion is done in cv2. Not sure why that happens

Reply:
Fixed now